components/codeblock/codeblock.stories.js

```diff
@@ -0,0 +1,40 @@
+import { createTemplateFromVueFile } from '@/common/storybook_utils';
+import DtCodeblock from './codeblock';
+import DtCodeblockMdx from './codeblock.mdx';
+import DtCodeblockDefaultTemplate from './codeblock_default.story.vue';
+
+export const argTypesData = {
+  // Props
+  text: {
+    description: 'The text to include in the codeblock',
+    control: {
+      type: 'text',
+    },
+  },
+};
+
+// Story Collection
+export default {
+  title: 'Components/Codeblock',
+  component: DtCodeblock,
+  argTypes: argTypesData,
+  excludeStories: /.*Data$/,
+  parameters: {
+    docs: {
+      page: DtCodeblockMdx,
+    },
+  },
+};
+
+// Templates
+const DefaultTemplate = (args, { argTypes }) => createTemplateFromVueFile(
+  args,
+  argTypes,
+  DtCodeblockDefaultTemplate,
+);
```

Comment:
One thing you have to change in vue 3 that you didn't really have any way of knowing, the templates don't have argTypes so this would be 

```
const DefaultTemplate = (args) => createTemplateFromVueFile(
  args,
  DtCodeblockDefaultTemplate,
);
```

Or else it won't render.